File: core/src/main/java/org/toml/ide/formatter/TomlFmtContext.java
/*
 * Use of this source code is governed by the MIT license that can be
 * found in the LICENSE file.
 */

package org.toml.ide.formatter;

import consulo.language.codeStyle.CodeStyleSettings;
import consulo.language.codeStyle.CommonCodeStyleSettings;
import consulo.language.codeStyle.SpacingBuilder;
import jakarta.annotation.Nonnull;
import org.toml.ide.formatter.impl.SpacingHelper;
import org.toml.lang.TomlLanguage;

public final class TomlFmtContext {
    private final CommonCodeStyleSettings commonSettings;
    private final SpacingBuilder spacingBuilder;

    public TomlFmtContext(@Nonnull CommonCodeStyleSettings commonSettings, @Nonnull SpacingBuilder spacingBuilder) {
        this.commonSettings = commonSettings;
        this.spacingBuilder = spacingBuilder;
    }

    @Nonnull
    public CommonCodeStyleSettings getCommonSettings() {
        return commonSettings;
    }

    @Nonnull
    public SpacingBuilder getSpacingBuilder() {
        return spacingBuilder;
    }

    @Nonnull
    public static TomlFmtContext create(@Nonnull CodeStyleSettings settings) {
        CommonCodeStyleSettings commonSettings = settings.getCommonSettings(TomlLanguage.INSTANCE);
        return new TomlFmtContext(commonSettings, SpacingHelper.createSpacingBuilder(commonSettings));
    }
}

File: core/src/main/kotlin/org/toml/ide/formatter/TomlFmtContext.kt
/*
 * Use of this source code is governed by the MIT license that can be
 * found in the LICENSE file.
 */

package org.toml.ide.formatter

import consulo.language.codeStyle.CodeStyleSettings
import consulo.language.codeStyle.CommonCodeStyleSettings
import consulo.language.codeStyle.SpacingBuilder
import org.toml.ide.formatter.impl.createSpacingBuilder
import org.toml.lang.TomlLanguage

data class TomlFmtContext(
    val commonSettings: CommonCodeStyleSettings,
    val spacingBuilder: SpacingBuilder
) {
    companion object {
        fun create(settings: CodeStyleSettings): TomlFmtContext {
            val commonSettings = settings.getCommonSettings(TomlLanguage)
            return TomlFmtContext(commonSettings, createSpacingBuilder(commonSettings))
        }
    }
}
